### DISCUSSION TITLE
use Controller to switch different field type with same field name

### DISCUSSION THREAD

**Discussion by chshouyu** [react-hook-form/react-hook-form#2243] [Q&A]
https://codesandbox.io/s/switch-between-different-fieldtype-with-same-fieldname-ckepy?file=/src/App.tsx

Suppose there is a form like the above, and suppose `input` and `select` are every complicated custom field, when I switch between two fields, the `select` can not get right results. when switch from input to select, there is a warn: 
```
Warning: The `value` prop supplied to <select> must be an array if `multiple` is true.
```

Expect:
1. when `fieldType` is `input`, the form data is
```json
{
  "name": "chen",
  "hobby": "play"
}
```
2. when `fieldType` is `select`, the form data is
```json
{
  "name": "chen",
  "hobby": ["run", "sleep"]
}
```

What should I do? thanks~

**Comment by bluebill1049** [comment]
try to toggle the `key` prop.

**Reply by chshouyu** [reply]
no, it is not work
